### PR TITLE
CO Vendor Tweaks

### PR DESCRIPTION
**Pull request by 4hands44**
# About the pull request

CO's only having one belt and satchel to choose from is weird and dumb. This PR seeks to Fix this by adding a variety of existing belts to the CO vendor, as well as touches the Weapon Vendor, by adding a variety attachments and equipment, to bring CO in line with SLs, IOs and other Field officer roles.

# Explain why it's good for the game
Gives COs more options for their loadouts

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Added more attachments, and belts to the CO arsenal.
add: CO now has an Essentials Kit like other roles, containing his Designator, and other useful tools.
add: Re-Adds Bridgecoat to some Officer Dress vendors. (Limited to CO(+) and XO currently.)
del: Removed Laser Designator from CO spawn Preset.
balance: CO can now vend welding Helmet visors.
balance: Adds grenade packets to CO Vendor.
/:cl:

### CODE REVIEW

File: code/game/machinery/vending/vendor_types/crew/commanding_officer.dm [hunk 1]
```diff
@@ -63,9 +71,18 @@ GLOBAL_LIST_INIT(cm_vending_gear_commanding_officer, list(
...
 		list("MRE", 0, /obj/item/storage/box/MRE, MARINE_CAN_BUY_MRE, VENDOR_ITEM_MANDATORY),
 
+		list("COMMANDING OFFICER ESSENTIALS KIT (TAKE ALL)", 0, null, null, null),
+		list("Commanding Officer essentials Kit", 0, /obj/effect/essentials_set/commanding_officer, MARINE_CAN_BUY_ESSENTIALS, VENDOR_ITEM_MANDATORY),
+
+		list("BAGS (CHOOSE 1)", 0, null, null, null),
+		list("Commanding Officer Backpack", 0, /obj/item/storage/backpack/mcommander, MARINE_CAN_BUY_BACKPACK, VENDOR_ITEM_MANDATORY),
+		list("Secure Satchel", 0, /obj/item/storage/backpack/satchel/lockable, MARINE_CAN_BUY_BACKPACK, VENDOR_ITEM_MANDATORY),
+		list("Expedition Pack", 0, /obj/item/storage/backpack/marine/satchel/intel, MARINE_CAN_BUY_BACKPACK, VENDOR_ITEM_MANDATORY),
+		list("M3 Sniper's Smock", 0, /obj/item/storage/backpack/marine/smock, MARINE_CAN_BUY_BACKPACK, VENDOR_ITEM_MANDATORY),
```
Comment (fira) [inline code/game/machinery/vending/vendor_types/crew/commanding_officer.dm:83]: Isn't that kind of out of place?

Comment (4hands44) [reply code/game/machinery/vending/vendor_types/crew/commanding_officer.dm:83]: Could you clarify?

File: code/game/machinery/vending/vendor_types/crew/commanding_officer.dm [hunk 2]
```diff
@@ -26,26 +26,34 @@ GLOBAL_LIST_INIT(cm_vending_gear_commanding_officer, list(
 		list("SPECIAL AMMUNITION", 0, null, null, null),
 		list("M41A Incendiary Magazine", 65, /obj/item/ammo_magazine/rifle/incendiary, null, VENDOR_ITEM_REGULAR),
 		list("M41A Rubber Shot Magazine", 10, /obj/item/ammo_magazine/rifle/rubber, null, VENDOR_ITEM_REGULAR),
+		list("HEDP Grenade Pack", 15, /obj/item/storage/box/packet/high_explosive, null, VENDOR_ITEM_REGULAR),
+		list("HEFA Grenade Pack", 15, /obj/item/storage/box/packet/hefa, null, VENDOR_ITEM_REGULAR),
+		list("WP Grenade Pack", 15, /obj/item/storage/box/packet/phosphorus, null, VENDOR_ITEM_REGULAR),
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/crew/commanding_officer.dm:31]: split these into their own category

File: code/game/machinery/vending/vendor_types/crew/commanding_officer.dm [hunk 3]
```diff
@@ -1,9 +1,9 @@
 //------------GEAR VENDOR---------------
 
 GLOBAL_LIST_INIT(cm_vending_gear_commanding_officer, list(
-		list("COMMANDING OFFICER'S PRIMARY (CHOOSE 1)", 0, null, null, null),
-		list("M46C pulse rifle", 0, /obj/effect/essentials_set/co/riflepreset, MARINE_CAN_BUY_ESSENTIALS, VENDOR_ITEM_MANDATORY),
-		list("M56C Smartgun", 0, /obj/item/storage/box/m56c_system, MARINE_CAN_BUY_ESSENTIALS, VENDOR_ITEM_MANDATORY),
+		list("COMMANDER'S PRIMARY (CHOOSE 1)", 0, null, null, null),
+		list("M46C pulse rifle", 0, /obj/effect/essentials_set/co/riflepreset,MARINE_CAN_BUY_SECONDARY, VENDOR_ITEM_MANDATORY),
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/crew/commanding_officer.dm:5]:
```suggestion
		list("M46C Pulse Rifle", 0, /obj/effect/essentials_set/co/riflepreset, MARINE_CAN_BUY_SECONDARY, VENDOR_ITEM_MANDATORY),
```

File: code/game/machinery/vending/vendor_types/crew/commanding_officer.dm [hunk 4]
```diff
@@ -63,9 +71,18 @@ GLOBAL_LIST_INIT(cm_vending_gear_commanding_officer, list(
 GLOBAL_LIST_INIT(cm_vending_clothing_commanding_officer, list(
 		list("STANDARD EQUIPMENT (TAKE ALL)", 0, null, null, null),
 		list("Headset", 0, /obj/item/device/radio/headset/almayer/mcom/cdrcom, MARINE_CAN_BUY_EAR, VENDOR_ITEM_MANDATORY),
-		list("Satchel", 0, /obj/item/storage/backpack/satchel/lockable, MARINE_CAN_BUY_BACKPACK, VENDOR_ITEM_MANDATORY),
 		list("MRE", 0, /obj/item/storage/box/MRE, MARINE_CAN_BUY_MRE, VENDOR_ITEM_MANDATORY),
 
+		list("COMMANDING OFFICER ESSENTIALS KIT (TAKE ALL)", 0, null, null, null),
+		list("Commanding Officer essentials Kit", 0, /obj/effect/essentials_set/commanding_officer, MARINE_CAN_BUY_ESSENTIALS, VENDOR_ITEM_MANDATORY),
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/crew/commanding_officer.dm:77]:
```suggestion
		list("Commanding Officer Essentials Kit", 0, /obj/effect/essentials_set/commanding_officer, MARINE_CAN_BUY_ESSENTIALS, VENDOR_ITEM_MANDATORY),
```

File: code/game/machinery/vending/vendor_types/crew/commanding_officer.dm (hunk 1 above)
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/crew/commanding_officer.dm:83]: remove this.

File: code/game/machinery/vending/vendor_types/crew/commanding_officer.dm [hunk 5]
```diff
@@ -84,15 +101,21 @@ GLOBAL_LIST_INIT(cm_vending_clothing_commanding_officer, list(
...
+		list("M276 Ammo Load Rig", 0, /obj/item/storage/belt/marine, MARINE_CAN_BUY_BELT, VENDOR_ITEM_RECOMMENDED),
+		list("M276 Holster Toolrig", 0, /obj/item/storage/belt/gun/utility/full, MARINE_CAN_BUY_BELT, VENDOR_ITEM_RECOMMENDED),
+		list("M276 M82F Holster Rig", 0, /obj/item/storage/belt/gun/flaregun, MARINE_CAN_BUY_BELT, VENDOR_ITEM_REGULAR),
 
 		list("POUCHES (CHOOSE 2)", 0, null, null, null),
 		list("First-Aid Pouch (Refillable Injectors)", 0, /obj/item/storage/pouch/firstaid/full, MARINE_CAN_BUY_POUCH, VENDOR_ITEM_REGULAR),
 		list("First-Aid Pouch (Splints, Gauze, Ointment)", 0, /obj/item/storage/pouch/firstaid/full/alternate, MARINE_CAN_BUY_POUCH, VENDOR_ITEM_REGULAR),
 		list("First-Aid Pouch (Pill Packets)", 0, /obj/item/storage/pouch/firstaid/full/pills, MARINE_CAN_BUY_POUCH, VENDOR_ITEM_RECOMMENDED),
-		list("Medical Kit Pouch", 0, /obj/item/storage/pouch/medkit, MARINE_CAN_BUY_POUCH, VENDOR_ITEM_REGULAR),
+		list("Medkit Pouch", 0, /obj/item/storage/pouch/medkit, MARINE_CAN_BUY_POUCH, VENDOR_ITEM_REGULAR),
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/crew/commanding_officer.dm:116]: revert this

File: code/game/machinery/vending/vendor_types/crew/commanding_officer.dm [hunk 6]
```diff
@@ -84,15 +101,21 @@ GLOBAL_LIST_INIT(cm_vending_clothing_commanding_officer, list(
...
 		list("Autoinjector Pouch", 0, /obj/item/storage/pouch/autoinjector/full, MARINE_CAN_BUY_POUCH, VENDOR_ITEM_REGULAR),
+		list("Command Pouch", 0, /obj/item/storage/pouch/pistol/command, MARINE_CAN_BUY_POUCH, VENDOR_ITEM_REGULAR),
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/crew/commanding_officer.dm:118]: remove this

File: code/modules/projectiles/guns/rifles.dm [hunk 7]
```diff
@@ -463,6 +463,7 @@
 		/obj/item/attachable/flashlight/grip,
 		/obj/item/attachable/stock/rifle/collapsible,
 		/obj/item/attachable/attached_gun/grenade,
+		/obj/item/attachable/attached_gun/grenade/mk1,
```
Comment (Zonespace27) [inline code/modules/projectiles/guns/rifles.dm:466]: remove this

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/game/machinery/vending/vendor_types/crew/commanding_officer.dm [hunk 8]
```diff
@@ -28,24 +28,34 @@ GLOBAL_LIST_INIT(cm_vending_gear_commanding_officer, list(
...
+		list("Welding Visor", 5, /obj/item/device/helmet_visor/welding_visor, null, VENDOR_ITEM_RECOMMENDED),
+
 		list("UNDERBARREL ATTACHMENTS", 0, null, null, null),
 		list("Laser Sight", 15, /obj/item/attachable/lasersight, null, VENDOR_ITEM_REGULAR),
 		list("Angled Grip", 15, /obj/item/attachable/angledgrip, null, VENDOR_ITEM_REGULAR),
 		list("Vertical Grip", 15, /obj/item/attachable/verticalgrip, null, VENDOR_ITEM_REGULAR),
 		list("Underbarrel Shotgun", 15, /obj/item/attachable/attached_gun/shotgun, null, VENDOR_ITEM_REGULAR),
 		list("Underbarrel Extinguisher", 15, /obj/item/attachable/attached_gun/extinguisher, null, VENDOR_ITEM_REGULAR),
 		list("Underbarrel Flamethrower", 15, /obj/item/attachable/attached_gun/flamer, null, VENDOR_ITEM_REGULAR),
+		list("Underbarrel Grenade Launcher", 15, /obj/item/attachable/attached_gun/grenade/mk1, null, VENDOR_ITEM_REGULAR),
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/crew/commanding_officer.dm:52]: exchange this for the regular GL instead of the mk1's

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Review by Zonespace27 — APPROVED** [review]

File: code/game/machinery/vending/vendor_types/crew/commanding_officer.dm [hunk 9]
```diff
@@ -28,24 +28,34 @@ GLOBAL_LIST_INIT(cm_vending_gear_commanding_officer, list(
...
+		list("HEFA Grenade Pack", 15, /obj/item/storage/box/packet/hefa, null, VENDOR_ITEM_REGULAR),
+		list("WP Grenade Pack", 15, /obj/item/storage/box/packet/phosphorus, null, VENDOR_ITEM_REGULAR),
+
 		list("RAIL ATTACHMENTS", 0, null, null, null),
 		list("Red-Dot Sight", 15, /obj/item/attachable/reddot, null, VENDOR_ITEM_REGULAR),
 		list("Reflex Sight", 15, /obj/item/attachable/reflex, null, VENDOR_ITEM_REGULAR),
 		list("S4 2x Telescopic Mini-Scope", 15, /obj/item/attachable/scope/mini, null, VENDOR_ITEM_REGULAR),
 
+		list("Helmet Visors", 0, null, null, null),
+		list("Night Vision Visor", 25, /obj/item/device/helmet_visor/night_vision, null, VENDOR_ITEM_RECOMMENDED),
```
Comment (Zonespace27) [inline code/game/machinery/vending/vendor_types/crew/commanding_officer.dm:42]:
```suggestion
```
Changed my mind about this

**Pull request merged**